ribs/archives/_cvt_archive.py

```diff
@@ -164,16 +166,55 @@ def __init__(self,
         self._chunk_size = chunk_size
 
         if custom_centroids is None:
-            if not isinstance(samples, int):
-                # Validate shape of custom samples. These are ignored when
-                # `custom_centroids` is provided.
-                samples = np.asarray(samples, dtype=self.dtype)
-                if samples.shape[1] != self._measure_dim:
-                    raise ValueError(
-                        f"Samples has shape {samples.shape} but must be of "
-                        f"shape (n_samples, len(ranges)={self._measure_dim})")
-            self._samples = samples
-            self._centroids = None
+            if centroid_method == "kmeans":
+                if not isinstance(samples, int):
+                    # Validate shape of custom samples. These are ignored when
+                    # `custom_centroids` is provided.
+                    samples = np.asarray(samples, dtype=self.dtype)
+                    if samples.shape[1] != self._measure_dim:
+                        raise ValueError(
+                            f"Samples has shape {samples.shape} but must be of "
+                            f"shape (n_samples, len(ranges)="
+                            f"{self._measure_dim})")
+                self._samples = samples
+                self._centroids = None
+                if self._centroids is None:
+                    self._samples = self._rng.uniform(
+                        self._lower_bounds,
+                        self._upper_bounds,
+                        size=(self._samples, self._measure_dim),
+                    ).astype(self.dtype) if isinstance(self._samples,
+                                                       int) else self._samples
+
+                    self._centroids = k_means(self._samples, self._cells,
+                                              **self._k_means_kwargs)[0]
+
+                    if self._centroids.shape[0] < self._cells:
+                        raise RuntimeError(
+                            "While generating the CVT, k-means clustering "
+                            f"found {self._centroids.shape[0]} centroids, "
+                            f"but this archive needs {self._cells} cells. "
+                            f"This most likely happened because there are "
+                            "too few samples and/or too many cells.")
+            elif centroid_method == "random":
+                # generates random centroids across two ranges
+                self._centroids = self._rng.uniform(self._lower_bounds,
+                                                    self._upper_bounds,
+                                                    size=(self._cells, 2))
```

Comment:
2 should be measure_dim I believe. Can you go through and make sure that all of these methods work for any measure dimension? Seems like most of them are hard-coded to 2D right now.